elements/bulbs-liveblog/bulbs-liveblog.js

```diff
@@ -0,0 +1,238 @@
+    };
+    let firebaseAppName = `liveblog-${this.getAttribute('liveblog-id')}`;
+
+    this.firebaseApp = firebase.initializeApp(firebaseAppConfig, firebaseAppName);
+    this.firebaseDatabase = this.firebaseApp.database();
+    this.firebaseRef = this.firebaseDatabase
+                        .ref(this.getAttribute('firebase-path'))
+                        .orderByChild('published')
+                        .limitToLast(100);
+  }
+
+  setupInterval () {
+    this.interval = setInterval(this.handleInterval, LIVEBLOG_LATENCY);
+  }
+
+  setupEvents () {
+    this.addEventListener('click', this.handleClick);
+    this.addEventListener('liveblog-entry-attached', this.handleEntryAttached);
+    this.addEventListener('liveblog-entry-detached', this.handleEntryDetached);
+    this.firebaseRef.on('value', this.handleFirebaseValue);
+  }
+
+  teardownFirebase () {
+    this.firebaseRef.off();
+  }
+
+  teardownInterval () {
+    clearInterval(this.interval);
+  }
+
+  handleFirebaseValue (snapshot) {
+    this.entriesData = snapshot.val() || {};
+    Object.keys(this.entriesData).forEach((entryId) => {
+      parseEntry(this.entriesData[entryId]);
+    });
+    this.handleInterval();
+  }
+
+  handleInterval () {
+    if (this.fetching) {
+      return;
+    }
+
+    let entryIds = this.getEntryIdsToFetch();
+    if (entryIds.length) {
+      this.handleBlogUpdate(entryIds);
+    }
+  }
+
+  handleEntryAttached () {
+    let thisEntry = {
+      element: event.target,
+      published: event.target.published,
+    };
+
+    this.entriesStore.all[event.target.getAttribute('entry-id')] = thisEntry;
+
+    if (!this.entriesStore.oldestEntryDate || thisEntry.published < this.entriesStore.oldestEntryDate) {
+      this.entriesStore.oldestEntryDate = thisEntry.published;
+    }
+  }
+
+  handleEntryDetached () {
+    delete this.entriesStore.all[this.getAttribute('entry-id')];
+  }
+
+  getEntryIdsToFetch () {
+    let now = new Date();
+    let entryIds = [];
+    Object.keys(this.entriesData).forEach((entryId) => {
+      let entry = this.entriesData[entryId];
+      if (entry.published && entry.published < now && !this.entriesStore.all[entryId]) {
+        if (this.entriesStore.oldestEntryDate) {
+          if (entry.published >= this.entriesStore.oldestEntryDate) {
+            entryIds.push(entryId);
+          }
+        }
+        else {
+          entryIds.push(entryId);
+        }
+      }
+    });
+    return entryIds;
+  }
+
+  handleClick (event) {
+    if (event.target.matches('button.liveblog-new-entries')) {
+      this.showNewEntries();
+      this.resetSelectedEntry();
+    }
+
+    if (event.target.matches('button.liveblog-entry-reset')) {
+      this.resetSelectedEntry();
+    }
+  }
+
+  showNewEntries () {
+    this.removeNewEntriesButton();
+    while (this.entryStaging.firstElementChild) {
+      this.entriesContainer[0].prepend(this.entryStaging.firstElementChild);
+    }
+    let header = document.querySelector('header');
```

Comment:
Could the header query selector be a configuration?

In it's current state, reliance on the `header` element is a hidden dependency, something that would be hard to find out if it wasn't working correctly for this reason.

If not a configuration, can this dependence be documented somewhere that's easily accessible if someone were to use this element?